drivers/t5838/t5838_aad_mode.c

```diff
@@ -0,0 +1,486 @@
+		{T5838_REG_AAD_D_ABS_THR_LO, aadconf->aad_d_abs_thr & 0xFF},
+		{T5838_REG_AAD_D_ABS_THR_HI, ((aadconf->aad_d_abs_thr >> 8) & 0x1F)},
+		{T5838_REG_AAD_D_REL_THR, aadconf->aad_d_rel_thr},
+		{T5838_REG_AAD_MODE, T5838_AAD_SELECT_D2},
+	};
+	err = prv_aad_mode_set(dev, write_data, ARRAY_SIZE(write_data));
+	if (err < 0) {
+		LOG_ERR("prv_aad_mode_set, err: %d", err);
+		return err;
+	}
+	drv_data->aad_enabled_mode = T5838_AAD_SELECT_D2;
+	return 0;
+}
+
+int t5838_aad_mode_disable(const struct device *dev)
+{
+
+	const struct t5838_aad_drv_cfg *drv_cfg = dev->config;
+	struct t5838_aad_drv_data *drv_data = dev->data;
+	struct t5838_drv_data *pdm_data = drv_cfg->pdm_dev->data;
+	int err;
+	/** Set AAD mode to zero while we configure device to avoid problems */
+	err = prv_reg_write(dev, T5838_REG_AAD_MODE, 0);
+	if (err < 0) {
+		LOG_ERR("prv_t5838_reg_write, err: %d", err);
+		return err;
+	}
+	/** Disable interrupts */
+	err = gpio_pin_interrupt_configure_dt(&drv_cfg->wake, GPIO_INT_DISABLE);
+	if (err) {
+		LOG_ERR("gpio_pin_interrupt_configure_dt, err: %d", err);
+		return err;
+	}
+	drv_data->aad_enabled_mode = T5838_AAD_SELECT_NONE;
+	pdm_data->aad_child_dev = NULL;
+	return 0;
+}
+
+/**
+ * @brief Function for initializing T5838 AAD trigger device. Called during device boot.
+ *
+ * @param[in] dev Pointer to the device structure for the driver instance.
+ */
+static int t5838_aad_init(const struct device *dev)
+{
+	const struct t5838_aad_drv_cfg *drv_cfg = dev->config;
+	int err;
+	/* T5838 specific configuration */
+
+	err = gpio_pin_configure_dt(&drv_cfg->wake, GPIO_INPUT);
+	if (err < 0) {
+		if (err < 0) {
+			LOG_ERR("gpio_pin_configure_dt, err: %d", err);
+			return err;
+		}
+	}
+
+	err = gpio_pin_configure_dt(&drv_cfg->thsel, GPIO_OUTPUT);
+	if (err < 0) {
+		if (err < 0) {
+			LOG_ERR("gpio_pin_configure_dt, err: %d", err);
+			return err;
+		}
+	}
+
+	err = gpio_pin_configure_dt(&drv_cfg->pdmclk, GPIO_OUTPUT);
+	if (err < 0) {
+		if (err < 0) {
+			LOG_ERR("gpio_pin_configure_dt, err: %d", err);
+			return err;
+		}
+	}
+
+	/* set micen low to turn of microphone and make sure we start in reset state*/
+	if (drv_cfg->micen_available) {
+		err = gpio_pin_configure_dt(&drv_cfg->micen, GPIO_OUTPUT);
+		if (err < 0) {
+			if (err < 0) {
+				LOG_ERR("gpio_pin_configure_dt, err: %d", err);
+				return err;
+			}
+		}
+		t5838_reset(dev);
+	}
+	/** if AAD mode is selected, first write unlock sequence */
+	err = prv_aad_unlock_sequence(dev);
+	if (err < 0) {
+		LOG_ERR("error writing aad unlock sequence, err: %d", err);
+		return err;
+	}
+	return 0;
+}
+
+#define T5838_AAD_DEVICE(idx)                                                                      \
+	static struct t5838_aad_drv_data t5838_aad_data_##idx;                                     \
+	static const struct t5838_aad_drv_cfg t5838_aad_cfg##idx = {                               \
+		.pdm_dev = DEVICE_DT_GET(DT_NODELABEL(pdm0)),                                      \
```

Comment:
I can try using the bus functions, I am however not sure if they actually work with PDM (I did try at some point and had some problems, not sure what it was exactly). Hard-coding that is however not really a problem since nRF always has only one PDM that should always be labeled pdm0.

I will look into this, it shouldn't be introducing any problems however (except perhaps in case of porting this driver to another architecture where that is not the case).